### ISSUE TITLE
Command Line support

### ISSUE THREAD

**Issue by Midnex** [rustdesk/rustdesk#69]
Would like to see command line support so I can add RustDesk to env variables, and push an id as an argument along with ability to push a password if needed.

Of course the command line interface would need to support other things as well, but this support would be a nice addition.

This feature could also be used to drop shortcuts on to the users desktop, that calls RustDesk and connects to a saved id in the shortcut file. Similar to how AnyDesk does it.

**Comment by rustdesk** [comment]
> drop shortcuts

1.1.6

**Comment by rustdesk** [comment]
https://support.anydesk.com/Command_Line_Interface

**Comment by rustdesk** [comment]
Because someone mentioned this on discord, and I have been investigating on AnyDesk/TeamViewer/Parsec API lately.

**Comment by bilogic** [comment]
Teamviewer https://community.teamviewer.com/English/kb/articles/34447-command-line-parameters

I normally use `teamviewer.exe -i <id>`, and it prompts for password, very useful if there are many machines to manage.

**Comment by rustdesk** [comment]
`--connect <id>`

**Comment by bilogic** [comment]
> `--connect <id>`

wow thanks! is it documented anywhere?

**Comment by Shahin-rmz** [comment]
Hello,
A power user (not a dev) here.
First:
Rustdesk is WAY better than Anydesk or Teamviewer.
I have installed it on my parent's Raspberry pi.
and can control the Raspberry remotely and play Movie for them.
Thanks <3
Now a question:
_________________________________________________________________
Can I connect via SSH and generate a password, (or see the password) and then connect via client to the Raustdesk?
Thanks

**Comment by bilogic** [comment]
> Can I connect via SSH and generate a password, (or see the password) and then connect via client to the Raustdesk?

Are you suggesting this is more secured than having a static complex passwordon rustdesk?

Frankly, I think it takes the same flow and effort to steal your static complex password vs your SSH password/key.

**Comment by Shahin-rmz** [comment]
> Frankly, I think it takes the same flow and effort to steal your static complex password vs your SSH password/key.


You are right.
Ultimately I like to:
-  have a dynamic password for **every new connection every time** but a static password for trusted connections.
Or other way around,
-  trusted connections make a pair of password each time.

**Comment by bilogic** [comment]
Haha, we are going in circles: how do you determine if a connection is a trusted one?

Teamviewer does it by having 2 passwords, a dynamic one and static one, either one will allow you to login.

**Comment by Shahin-rmz** [comment]
> Haha, we are going in circles: how do you determine if a connection is a trusted one?

You are right. :) I am thinking that if both parties *check the trusted box*, then we the connections are *trusted*.

> The closest I can think of is to have the dynamic password regenerate after each disconnection, but the static remains and is available to all connections, just make sure to keep it complex.

This is one good approach, the other one is to have static password just for some connections and **not global**.

**Comment by paulbdavis** [comment]
Jumping in this issue to say it would be good to be able to get the current ID and password via the command line. 

The use case is having a machine that's out in the field where there is not someone present to read off the ID and password or interact with the UI component (the UI would not even be launched). It would be good to dump these out on the command line when connected to said machine via SSH

like:

```
$ rustdesk --info

ID: 123 123 123
Password: abcdef
```

**Comment by paspo** [comment]
you can read ID and password from the config files:

- C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml (windows)
- /root/.config/rustdesk/RustDesk.toml (linux)
- /Library/Preferences/com.carriez.RustDesk/RustDesk.toml (OSX)

**Comment by rustdesk** [comment]
will be encrypted in next release

**Comment by bilogic** [comment]
> Jumping in this issue to say it would be good to be able to get the current ID and password via the command line.

I think it is generally not safe to expose the ID and password via an "API" approach. Anything else can also call the "API" and gain access. Getting a dynamic password via "API" using SSH is equivalent to setting a static password (SSH being the static in this case).

It is probably best to just let Rust Desk do its best to protect the static password instead of making it leak credentials via an "API".

**Comment by paulbdavis** [comment]
For the password, sure, I suppose that makes sense, but getting the ID and being able to *set* the password would be enough really.

Having it disabled by default would obviously be best as well, but there should be a way to get/set the credentials for a remote machine without anyone actually being present.

**Comment by bilogic** [comment]
> Having it disabled by default would obviously be best as well, but there should be a way to get/set the credentials for a remote machine without anyone actually being present.

In that case, I think it is better to state the problem ("How to perform unattended remote installation") instead of suggesting the solution.

In one of the VPS that I use, they ask for my SSH public key when provisioning the server. i.e. unattended installs must have a static password, but please don't quote me on that, authentication/credential modifications must always be rigorously thought out.

**Comment by jcelerier** [comment]
>  In that case, I think it is better to state the problem ("How to perform unattended remote installation") instead of suggesting the solution.

what's the solution to this problem? how can I get a remote desktop remotely from a machine which has nothing installed and to which I have a ssh access ?

**Comment by paulbdavis** [comment]
Sure, for the password that is fine, I can just set it manually, but what about getting the ID? Will that also be encrypted on the next release?

**Comment by rustdesk** [comment]
Yes， enced in next release.

**Comment by ssteeltm** [comment]
would be nice if I could connect via CMD with:
`` rustdesk.exe --connect <ID> --password <password> ``

**Comment by bilogic** [comment]
`--connect <id>`, use rustdesk to remember the password

**Comment by Onyx47** [comment]
Lack of an option to retrieve the ID programmatically, possibly from a file or as a command-line option that can only be read/ran as root/elevated user is what's currently stopping me from replacing AnyDesk with RustDesk for my use case.

Having the ID recorded somewhere where it's only accessible to root/admin does not strike me as a huge security risk (AnyDesk does this as well, as an aside), and enables easier automation for cases where you already have some kind of elevated access available to you and you just need that extra step of being able to access the GUI.

Also, just to point out, when starting RustDesk from terminal on Linux, it still spits out the ID to `stdout`, so it's technically still possible to grab the ID programatically, but this is not something I'd rely on, plus it's inconsistent - it's encrypted in a file but it's readable by a regular user if they can just execute a command as an unprivileged user.

**Comment by mikelwellsmore** [comment]
Agreed - we need a way to programmatically retrieve the ID and set the permanent password.

**Comment by kapytanhook** [comment]
we need to be able to set or read the ID or there is no point to this software, going to have to make my own build just to do that now.

**Comment by Onyx47** [comment]
For anyone needing this functionality and is willing to build their own release, this is my current solution until there is a more definite decision on this: https://github.com/Onyx47/rustdesk/commit/ed9e92fc2e46fb3272cb4d694802ac8915a45ba7

It enables you to run `rustdesk --id` as admin/root and simply prints out the ID and quits. Certainly not sophisticated but it should do for scripting purposes. Only tested on Linux but I'd assume Windows should be fine as well, it's using the same logic as `--password` argument.

**Comment by knro** [comment]
It would be nice to be able to set/get ID as root. Any ideas if this will ever be merged in Rustdesk or we'd have to rollout our own releases?

**Comment by merlink01** [comment]
For autologin the way anydesk does it is nice:
Load the password from stdin.

So the command is:
`rustdesk.exe --connect <ID> --password`
and after this:
send the password via stdin to the process followed ba an \n.

On this way the password is not visible on the process list.

**Comment by gmdig** [comment]
> Highly recommend this, I can only precompile one client per client right now, built in id and password ,is there any body can share a document ,how to pre-config it

Is it possible to replace your built in id and password with environment variables?
